### ISSUE TITLE
TiShadow doesn't load any images on iOS but does on Android

### ISSUE THREAD

**Issue by cdrx** [dbankier/TiShadow#423]
Using either --shadow or appify TiShadow doesn't load images for me on iOS. 

Running the same project with "-p android --shadow" does load the images.

I'm using Titanium CLI version 3.4.2, Titanium SDK version 3.5.1.GA, TiShadow version 2.8.1, all freshly installed from npm today. It worked before upgrading.

It affects all the projects I've got, even an empty project with just a single ImageView.

I get these errors when running --shadow on my app with iOS

[ERROR] Cannot remove property "ti.android.bug2373.finishfalseroot", it is read-only.
[ERROR] Cannot remove property "ti.android.bug2373.skipAlert", it is read-only.
[ERROR] Cannot remove property "ti.ui.defaultunit", it is read-only.
[ERROR] 2015-03-21 15:26:21.858 App[49958:14273876] 1803 entries in the zip file

If I look at the file created in dist/tishadow/xx.zip I can see all the assets in the file.

Any idea how I can go about debugging this? I'm a bit stumped.

**Comment by karaoak** [comment]
I am experiencing the exact same issue as @cdrx describes. Same Ti SDK, same TiShadow version.

**Comment by cdrx** [comment]
@karaoak what OS X version?

**Comment by dbankier** [comment]
Can you enter `cat ~/.titanium/config.json` and share with me the plugins section?
Just a hunch...

**Comment by cdrx** [comment]
There is nothing in my plugins section, the whole file looks like this:

```
{
	"user": {
		"locale": "en_US"
	},
	"app": {
		"workspace": ""
	},
	"cli": {
		"colors": true,
		"completion": false,
		"logLevel": "trace",
		"prompt": true,
		"progressBars": true,
		"failOnWrongSDK": false,
		"httpProxyServer": "",
		"rejectUnauthorized": true,
		"width": 100,
		"ignoreDirs": "^(\\.svn|_svn|\\.git|\\.hg|\\.?[Cc][Vv][Ss]|\\.bzr|\\$RECYCLE\\.BIN)$",
		"ignoreFiles": "^(\\.gitignore|\\.npmignore|\\.cvsignore|\\.DS_Store|\\._.*|[Tt]humbs.db|\\.vspscc|\\.vssscc|\\.sublime-project|\\.sublime-workspace|\\.project|\\.tmproj)$"
	},
	"paths": {
		"commands": [],
		"hooks": [
			"/usr/local/lib/node_modules/tishadow/plugin/yy.tishadow/1.0/hooks"
		],
		"modules": [],
		"plugins": [],
		"sdks": [
			"/Volumes/Data 1/cdr/Library/Application Support/Titanium"
		],
		"templates": []
	},
	"sdk": {
		"defaultInstallLocation": "/Volumes/Data 1/cdr/Library/Application Support/Titanium/",
		"selected": "3.5.1.GA"
	},
	"android": {
		"sdkPath": "/Volumes/Data 1/cdr/Projects/android-sdk"
	}
```

**Comment by dbankier** [comment]
Sorry, I meant hooks, but it looks fine.
Can you send me a basic sample what doesn't work for you.
The issue seems to sound like #385 or #394  (but they are fixed).

**Comment by cdrx** [comment]
```
var win1 = Titanium.UI.createWindow({  
    backgroundColor:'#fff'
});
var image = Ti.UI.createImageView({
	image: 'KS_nav_ui.png'
});
win1.add(image);
win1.open();
```

This works for me with TiShadow built for Android but not when built for iOS. 

Building it for iOS without --shadow does show the image correctly.

**Comment by shodanuk** [comment]
Out of curiosity, if you move your image to assets/images and change the image param of your ImageView accordingly, does that change anything?

**Comment by cdrx** [comment]
Just tried that, doesn't work either I'm afraid.

**Comment by dbankier** [comment]
Hate to say this as it is never helpful.
I just tried your code and "it worked for me".
ts 2.8.1
tisdk 3.5.1.GA
ticli 3.5.1 & 3.5.2
iOS 8.2 (sim)

Not sure if I can be much more help. My only guess left is that you have multiple installs of ts.

**Comment by cdrx** [comment]
I've uninstalled Titanium, node, npm, tishadow, stripped everything right back and reinstalled clean, still getting the same issue.

The only thing that I know is different about my setup is that I'm running 10.10.3 beta.

Does TiShadow have a log or some sort of debug mode I can put it in to get a clearer idea of what its doing?

**Comment by dbankier** [comment]
Next thing is for you to share your `build/tishadow/src/app.js` file that `ts` the sample app app.js to.
And `build/appify/Resources/api/PlatformRequire.js`.

**Comment by cdrx** [comment]
I've put both files in this gist: https://gist.github.com/cdrx/da27426213ebe6c31dc2

**Comment by WooD1k** [comment]
Same happens to me.
w/o --shadow everything is working as expected okay

here is my hooks section from config.json:

```json
"hooks": [
			"/usr/local/lib/node_modules/titanium-code-processor/hooks",
			"/usr/local/lib/node_modules/tishadow/plugin/yy.tishadow/1.0/hooks",
			"/Applications/Appcelerator Studio/plugins/com.appcelerator.titanium.liveview.core_1.1.0.1424300841/node_modules/liveview/hook"
		],
```
```bash
$ tishadow -v
2.8.1
```

OS X version - 10.10.3 (14D127a)
What can i do to help you to 'detect' this issue?

Thanks in advance!

**Comment by chmiiller** [comment]
I'd this problem with invisible images for iOS but without the ERRORs logs. I fixed it, juts putting the images without the @2x sign.
In the app I only had the "image@2x.png" and it doesn't work. Then I created the "image.png" with half size and kept the "image@2x.png" file.
I really don't understand why some Icons and images are shown smaller on Android =/

**Comment by dbankier** [comment]
@chmiller, thanks for finding the issue, a fix is on the way.
For your android one see #180 - it is an AppC issue and is fixed in 4.0.
